wapitiCore/attack/mod_network_device.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+from typing import Optional
+
+from bs4 import BeautifulSoup
+from httpx import RequestError
+
+from wapitiCore.attack.attack import Attack
+from wapitiCore.net import Request
+from wapitiCore.net.response import Response
+from wapitiCore.definitions.fingerprint import NAME as TECHNO_DETECTED, WSTG_CODE
+
+from wapitiCore.main.log import log_blue
+
+MSG_TECHNO_VERSIONED = "{0} {1} detected"
+MSG_NO_UBIKA = "No UBIKA Detected"
+
+
+class ModuleNetworkDevice(Attack):
+    """Base class for detecting version."""
+    name = "network_device"
+    version = ""
+
+    async def check_ubika(self, url):
+        check_list = ['app/monitor/']
+        for item in check_list:
+            request = Request(f'{url}{item}', 'GET')
+            try:
+                response: Response = await self.crawler.async_send(request, follow_redirects=True)
+            except RequestError:
```

Comment:
Your function does not return anything if there is a `RequestError` because catched by the try/except.
Consider returning a False by default or raising the same exception.